src/entities/SubscriptionSet.ts

```diff
@@ -0,0 +1,99 @@
+import { Subscription } from './Subscription';
+import { SubscribeCapable, SubscriptionOptions } from './commonTypes';
+
+export class SubscriptionSet implements SubscribeCapable {
+  private channelNames: string[] = [];
+  private groupNames: string[] = [];
+  private options: SubscriptionOptions;
+  private pubnub: any;
+  private eventEmitter: any;
+  private subscriptionList: Subscription[];
+
+  constructor({
+    channels = [],
+    channelGroups = [],
+    subscriptionOptions,
+    eventEmitter,
+    pubnub,
+  }: {
+    channels: string[];
+    channelGroups: string[];
+    subscriptionOptions: SubscriptionOptions;
+    eventEmitter: any;
+    pubnub: any;
+  }) {
+    this.channelNames = [...this.channelNames, ...channels];
+    this.groupNames = [...this.groupNames, ...channelGroups];
+    this.options = subscriptionOptions;
+    this.eventEmitter = eventEmitter;
+    this.pubnub = pubnub;
+    this.subscriptionList = [
+      new Subscription({
+        channels: this.channelNames,
+        channelGroups: this.groupNames,
+        subscriptionOptions: this.options,
+        eventEmitter: this.eventEmitter,
+        pubnub: this.pubnub,
+      }),
```

Comment:
This a bit unexpected. `Subscription` object is able to deal with only one entity and it's channel names. `this.subscriptionList` should store list of `Subscription` entities created for each provided `channel` / `group` entity.

Reply:
Is it like we don't need to keep reference list of `Subscription`s

Reply:
At least it is this way in other SDKs… they keep a list of Subscription instances, not a single subscription with multiple channels in it. Each subscription responsible only for own entity and tell whether it should handle provided events or not.

Reply:
Oh!! I think I added it to cover one usecase like.

what if I want one subscription  with 3 channels and 2 channelGroups.

this can be short way to do it instead creating all 3 channels and 2 channelGroups.

We can remove this initialisation if we want to consistent across SDKs.

e.g :

`pubnub.subscriptionSet({ channels: ['ch3', 'ch4'] });`

Reply:
But we can make those internally and store in the list of `Subscription`s

Reply:
Refactored here.

Keeping channels/groups set as class level private field so that We don't need to ask every subscription for their channelNames/GroupNames when any update(add/remove) apis will be called.